presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java

```diff
@@ -97,7 +97,7 @@ public void configure(Binder binder)
 
         binder.bind(HdfsConfigurationInitializer.class).in(Scopes.SINGLETON);
         newSetBinder(binder, DynamicConfigurationProvider.class);
-        binder.bind(DirectoryLister.class).to(CachingDirectoryLister.class).in(Scopes.SINGLETON);
+        binder.bind(DirectoryLister.class).annotatedWith(ForCachingDirectoryLister.class).to(HadoopDirectoryLister.class).in(Scopes.SINGLETON);
```

Comment:
nit: Maybe move this down to the changes below to keep `DirectoryLister` binding code together?